### PR TITLE
feat: add C implementation for `dist/triangular/median`

### LINKED ISSUE #3817: [RFC]: Add C implementation for `@stdlib/stats/base/dists/triangular/median`
### Description

This RFC proposes adding a C implementation, including a Node.js native addon, C benchmarks, and C examples, for the [`@stdlib/stats/base/dists/triangular/median`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/stats/base/dists/triangular/median) package.

When adding support, the following tasks should be completed:

1. A C source implementation should be added to a `src` folder within the package's directory.
2. A native add-on interface should be added in order to call the C implementation from JavaScript.
3. A JavaScript file should be added to the `lib` folder to allow benchmarking and unit testing the native implementation from JavaScript.
4. A new test file should be created which includes unit tests testing against expected behavior.
5. C benchmarks should be added which measure performance for the C source implementation.
6. A file containing C examples should be added to demonstrate example usage.
7. JavaScript benchmarks should be added which measure performance for the native add-on.
8. Build and configuration files should be added to allow compiling and running the C source implementation using project `make` commands (as documented below).
9. The README should be updated to include documentation for the added C API.

To provide a concrete example of what a PR adding the desired method should contain, see https://github.com/stdlib-js/stdlib/pull/3354, which is a PR adding a C implementation for the CDF of an arcsine distribution. This should provide an idea of what is expected.

#### Prerequisites

1. Review JavaScript implementation in `lib/main.js` and its required modules to identify all `stdlib` functions used.
2. Verify C implementations exist for each required function.

### Related Issues

#3355

### Questions

No.

### Other

Once the implementation is ready including C examples and benchmarks, we should be able to run the following `make` commands.

#### Build native add-on

```bash
NODE_ADDONS_PATTERN="@stdlib/stats/base/dists/triangular/median" make install-node-addons
```

#### Run C examples

```bash
make examples-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/triangular/median/examples/c/example.c"
```

#### Run C benchmarks

```bash
make benchmark-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/triangular/median/benchmark/c/benchmark.c"
```

#### Run JavaScript benchmarks

```bash
make benchmark-javascript-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/triangular/median/benchmark/benchmark.native.js"
```

#### Run all tests

```bash
make TESTS_FILTER=".*/@stdlib/stats/base/dists/triangular/median/.*" test
```

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please minimize the submission of LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by sethdivyansh**
Resolves #3817

## Description

- adds C implementation for @stdlib/stats/base/dists/triangular/median along with relevant tests, benchmarks and examples

This pull request:

-  progresses: #3817 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/base/dists/triangular/median/README.md
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/triangular/median/README.md]:
```suggestion
#### stdlib_base_dists_triangular_median( a, b, c )
```

File: lib/node_modules/@stdlib/stats/base/dists/triangular/median/README.md
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/triangular/median/README.md:211]: Please define the below here 

```c
static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v*(max-min) );
}
```

and use it to draw the random values for `a`, `b`, and `c` similar to the changes made in https://github.com/stdlib-js/stdlib/pull/3354/commits/da14b77af9c95ec5185a6aced49030f53d97fdee.

File: lib/node_modules/@stdlib/stats/base/dists/triangular/median/src/main.c
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/triangular/median/src/main.c]: This is not how we do our C documentation comments. Please consult the reference implementation or other merged PRs.

File: lib/node_modules/@stdlib/stats/base/dists/triangular/median/lib/native.js
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/triangular/median/lib/native.js]:
```suggestion
*
* @private
* @param {number} a - minimum support
```

File: lib/node_modules/@stdlib/stats/base/dists/triangular/median/examples/c/example.c
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/triangular/median/examples/c/example.c:22]: Same comment as in the `README.md`.

File: lib/node_modules/@stdlib/stats/base/dists/triangular/median/benchmark/c/benchmark.c
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/triangular/median/benchmark/c/benchmark.c]: Same comment as in the `README.md`.

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Thanks for your work!

Left a few comments that will need to be addressed before this PR can be merged; hopefully should be quick to rectify.

**Comment by sethdivyansh** [comment]
> Thanks for your work!
> 
> Left a few comments that will need to be addressed before this PR can be merged; hopefully should be quick to rectify.

Done with the changes. Please review.

File: lib/node_modules/@stdlib/stats/base/dists/triangular/median/examples/c/example.c
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/triangular/median/examples/c/example.c]:
```suggestion
	double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
	return min + ( v*(max-min) );
```

File: lib/node_modules/@stdlib/stats/base/dists/triangular/median/src/main.c
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/triangular/median/src/main.c]:
```suggestion
* Returns the median of a triangular distribution.
```

**Review by Planeshifter — APPROVED** [review]
Thanks for making the edits! Will merge shortly after CI is all green.

**Pull request merged**
